zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/AbstractArchiverJob.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.archiver;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractArchiverJob implements ArchiverJob {
+
+  public static final String DATES_AGG = "datesAgg";
+  public static final String INSTANCES_AGG = "instancesAgg";
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractArchiverJob.class);
+
+  protected final ScheduledExecutorService archiverExecutor;
+
+  private final BackoffIdleStrategy idleStrategy;
+  private final BackoffIdleStrategy errorStrategy;
+  private boolean shutdown = false;
+  private final double rolloverBatchSize;
+  private final long delayBetweenRuns;
+
+  public AbstractArchiverJob(final ScheduledExecutorService archiverExecutor) {
+    this.archiverExecutor = archiverExecutor;
+    idleStrategy = new BackoffIdleStrategy(2_000, 1.2f, 60_000);
+    errorStrategy = new BackoffIdleStrategy(100, 1.2f, 10_000);
+
+    rolloverBatchSize = 100; // TODO operateProperties.getArchiver().getRolloverBatchSize();
+    delayBetweenRuns =
+        Duration.ofMinutes(1)
+            .toMillis(); // TODO: operateProperties.getArchiver().getDelayBetweenRuns();
+  }
+
+  @Override
+  public void run() {
```

Comment:
What I thought was whether we also could trigger the archiver from the Exporter (without having the searching and retrigger approach). E.g if we reach a PROCESS complete event or something 🤔 But I guess this would make the solution more the complex?

Reply:
It is more complex. We don't want to block exporter thread. So the archiver has to run in background or async. So, how do you ensure archiving is triggered again, incase the exporter closed before archiving completes, and the new leader do not re-export the record. There would be several such "new" cases that we should consider. Given the time constraint, I would like to keep it simple and just port what we have already.

Reply:
Was more an afterthought (improvement) nothing I would do now.

Reply:
> What I thought is that the exporter can submit a job on the other thread pool to archive the respective instance.

Ya. Same issue as I mentioned before. There are several edge-cases to consider.